playbooks/common/openshift-cluster/initialize_facts.yml

```diff
@@ -8,3 +8,5 @@
       role: common
       local_facts:
         hostname: "{{ openshift_hostname | default(None) }}"
+  - set_fact:
+      openshift_docker_hosted_registry_network: "{{ hostvars[groups.oo_first_master.0].openshift.common.portal_net }}"
```

Comment:
@abutcher @detiber when review kicks off I wanna call this out specifically for attention. I ran into the problem where docker restarts multiple times in a config re-run because in using the docker role, if you forget to pass this var explicitly, the config changes. 

I don't really know what the right solution here is, I want defaults with a little logic for inventory vars, if they're not defined ideally they should be set to a default so we can just assume they're usable. That default might be programatic, and it might be based off another system like the first master, so role defaults don't feel quite right. 

My thought was we should have snippets we can include in playbooks that do all variable initialization, this facts file seemed like a decent first start.

I'd like to scale something like this up to cover a lot of other vars. I've found it really helpful in this branch to know based on my behavior of openshift_version role, that openshift_version + openshift_pkg_version + openshift_image_tag are all set and correct and can be used accordingly.

Maybe it's even a role, openshift_defaults...